mods/d2k/maps/harkonnen-07/harkonnen07.lua

```diff
@@ -0,0 +1,279 @@
+		{ "combat_tank_a", "combat_tank_a", "quad", "trike", "trike" },
+		{ "trooper", "trooper", "trooper", "quad", "quad" }
+	},
+
+	hard =
+	{
+		{ "trooper", "trooper", "trooper", "trooper", "trooper", "light_inf", "light_inf", "light_inf", "light_inf", "light_inf", "light_inf", "light_inf" },
+		{ "combat_tank_a", "combat_tank_a", "quad", "quad", "trike", "trike" },
+		{ "trooper", "trooper", "trooper", "trooper", "quad", "quad" }
+	}
+}
+
+InitialCorrinoReinforcements = { "trooper", "trooper", "trooper", "trooper", "trooper" }
+
+AtreidesPath = { AtreidesEntry1.Location, AtreidesRally1.Location }
+
+CorrinoPaths =
+{
+	{ CorrinoEntry1.Location, CorrinoRally1.Location },
+	{ CorrinoEntry1.Location, CorrinoRally2.Location }
+}
+
+InitialAtreidesPaths =
+{
+	{ AtreidesEntry2.Location, AtreidesRally2.Location },
+	{ AtreidesEntry3.Location, AtreidesRally3.Location },
+	{ AtreidesEntry4.Location, AtreidesRally4.Location }
+}
+
+InitialCorrinoPath = { CorrinoEntry2.Location, CorrinoRally3.Location }
+
+HarkonnenReinforcements =
+{
+	easy =
+	{
+		{ "trooper", "trooper", "light_inf", "light_inf", "light_inf", "light_inf", "light_inf" },
+		{ "quad", "trike", "trike", "siege_tank", "combat_tank_h", "combat_tank_h" }
+	},
+
+	normal =
+	{
+		{ "trooper", "trooper", "light_inf", "light_inf", "light_inf", "light_inf" },
+		{ "quad", "trike", "siege_tank", "combat_tank_h", "combat_tank_h" }
+	},
+
+	hard =
+	{
+		{ "trooper", "trooper", "light_inf", "light_inf", "light_inf" },
+		{ "quad", "trike", "trike", "combat_tank_h", "combat_tank_h" }
+	},
+}
+
+HarkonnenPaths = 
+{
+	{ HarkonnenEntry1.Location, HarkonnenRally1.Location },
+	{ HarkonnenEntry2.Location, HarkonnenRally2.Location }
+}
+
+SendHarkonnenReinforcements = function(number)
+	Reinforcements.ReinforceWithTransport(player, "carryall.reinforce", HarkonnenReinforcements[Difficulty][number], HarkonnenPaths[number], { HarkonnenPaths[number][1] })
+	Trigger.AfterDelay(DateTime.Seconds(9), function()
+		Media.PlaySpeechNotification(player, "Reinforce")
+	end)
+end
+
+SendEnemyReinforcements = function(player, delay, path, unitTypes, customCondition, customHuntFunction)
+	Trigger.AfterDelay(delay, function()
+		if customCondition and customCondition() then
+			return
+		end
+
+		local units = Reinforcements.ReinforceWithTransport(player, "carryall.reinforce", unitTypes, path, { path[1] })[2]
+
+		if not customHuntFunction then
+			customHuntFunction = IdleHunt
+		end
+		Utils.Do(units, customHuntFunction)
+
+		SendEnemyReinforcements(player, delay, path, unitTypes, customCondition, customHuntFunction)
+	end)
+end
+
+SendAirStrike = function()
+	if HiTechIsDead then
+		return
+	end
+
+	local targets = Utils.Where(player.GetActors(), function(actor)
+		return
+			actor.Type == "construction_yard" or
+			actor.Type == "barracks" or
+			actor.Type == "refinery" or
+			actor.Type == "light_factory" or
+			actor.Type == "heavy_factory" or
+			actor.Type == "starport" or
+			actor.Type == "research_centre"
```

Comment:
Doesn't `actor.HasProperty("Building")` work here? (Would kind of shorten this a lot.)

Reply:
Building doesn't look like i property. I can check for Sell, as if it sellable, it is ceetainly a building.